src/mo_frontend/typing.ml

```diff
@@ -1201,20 +1207,25 @@ let error_bin_op env at t1 t2 =
     display_typ_expand t1
     display_typ_expand t2
 
-let compare_pat_field (pf1 : pat_field) (pf2 : pat_field) =
-  compare pf1.it.id.it pf2.it.id.it
+let compare_pat_field pf1 pf2 = match pf1.it, pf2.it with
```

Comment:
This has to agree with the order of Type.compare_field for the typechecker to work, right? Maybe check it does and add a comment.